apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -245,6 +245,60 @@ export class AppRoomBridge extends RoomBridge {
 		return users.map((user: ICoreUser) => userConverter.convertToApp(user));
 	}
 
+	protected async getUnreadByUser(roomId: string, userId: string, options: GetMessagesOptions, appId: string): Promise<Array<IMessageRaw>> {
+		this.orch.debugLog(`The App ${appId} is getting the unread messages for the user: "${userId}" in the room: "${roomId}"`);
+
+		const messageConverter = this.orch.getConverters()?.get('messages');
+		if (!messageConverter) {
+			throw new Error('Message converter not found');
+		}
+
+		const room = await Rooms.findOneById(roomId, { projection: { _id: 1 } });
+
+		if (!room) {
+			throw new Error('Room not found');
+		}
+
+		const subscription = await Subscriptions.findOneByRoomIdAndUserId(roomId, userId, { projection: { ls: 1 } });
+
+		const lastSeen = subscription?.ls;
+		if (!lastSeen) {
+			return [];
+		}
+
+		const sort: Sort = options.sort?.createdAt ? { ts: options.sort.createdAt } : { ts: 1 };
+
+		const cursor = Messages.findVisibleByRoomIdBetweenTimestampsNotContainingTypes(roomId, lastSeen, new Date(), [], {
```

Comment:
This would return system messages as well as "unread". Is that a valid use case? 👀

Reply:
Yes it is thank you!